jetty-ee9/jetty-ee9-tests/jetty-ee9-test-sessions/jetty-ee9-test-sessions-common/src/test/java/org/eclipse/jetty/ee9/session/IdleSessionTest.java

```diff
@@ -246,7 +255,69 @@ public void testNullSessionCache() throws Exception
         }
         finally
         {
-            _server1.stop();
+            sessionTestSupport.stop();
+        }
+    }
+
+    public static class RequestListenerWrapper extends HttpStream.Wrapper
+    {
+        CountDownLatch _countDownLatch;
+        
+        public RequestListenerWrapper(HttpStream wrapped, CountDownLatch countDownLatch)
+        {
+            super(wrapped);
+            _countDownLatch = countDownLatch;
+        }
+
+        /**
+         * 
+         */
+        @Override
+        public void succeeded()
+        {
+            if (_countDownLatch != null)
+                _countDownLatch.countDown();
+            super.succeeded();
+        }
+
+        /**
+         * @param x the reason for the operation failure 
+         */
+        @Override
+        public void failed(Throwable x)
+        {
+            if (_countDownLatch != null)
+                _countDownLatch.countDown();
+            super.failed(x);
+        }
+    }
+    
+    public static class RequestListenerHandler extends Handler.Wrapper
+    {
+        CountDownLatch _countDownLatch;
+        
+        /**
+         * @param request the HTTP request to handle 
+         * @param response the HTTP response to handle
+         * @param callback the callback to complete when the handling is complete
+         * @return
+         * @throws Exception
+         */
+        @Override
+        public boolean handle(org.eclipse.jetty.server.Request request, Response response, Callback callback) throws Exception
+        {
+            request.addHttpStreamWrapper(s -> new RequestListenerWrapper(s, _countDownLatch));
```

Comment:
This could be replaced with the following:
```java
org.eclipse.jetty.server.Request.addCompletionListener(request, x ->
{
    if (_countDownLatch != null)
        _countDownLatch.countDown();
});
```

which makes the `RequestListenerWrapper` class not needed anymore.